<h1>pandas.errors.UnsortedIndexError</h1> <dl class="py exception"> <dt class="sig sig-object py" id="pandas.errors.UnsortedIndexError"> <em class="property"><span class="pre">exception</span></em><span class="sig-prename descclassname"><span class="pre">pandas.errors.</span></span><span class="sig-name descname"><span class="pre">UnsortedIndexError</span></span><a class="reference external" href="https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/errors/__init__.py#L91-L115"><span class="viewcode-link"><span class="pre">[source]</span></span></a>
</dt> <dd>
<p>Error raised when slicing a MultiIndex which has not been lexsorted.</p> <p>Subclass of <cite>KeyError</cite>.</p> <p class="rubric">Examples</p> <div class="doctest highlight-default notranslate">
<div class="highlight"><pre data-language="python">&gt;&gt;&gt; df = pd.DataFrame({"cat": [0, 0, 1, 1],
...                    "color": ["white", "white", "brown", "black"],
...                    "lives": [4, 4, 3, 7]},
...                   )
&gt;&gt;&gt; df = df.set_index(["cat", "color"])
&gt;&gt;&gt; df
            lives
cat  color
0    white    4
     white    4
1    brown    3
     black    7
&gt;&gt;&gt; df.loc[(0, "black"):(1, "white")]
Traceback (most recent call last):
UnsortedIndexError: 'Key length (2) was greater
than MultiIndex lexsort depth (1)'
</pre></div> </div> </dd>
</dl><div class="_attribution">
  <p class="_attribution-p">
    &copy; 2008&ndash;2022, AQR Capital Management, LLC, Lambda Foundry, Inc. and PyData Development Team<br>Licensed under the 3-clause BSD License.<br>
    <a href="https://pandas.pydata.org/pandas-docs/version/2.2.2/reference/api/pandas.errors.UnsortedIndexError.html" class="_attribution-link">https://pandas.pydata.org/pandas-docs/version/2.2.2/reference/api/pandas.errors.UnsortedIndexError.html</a>
  </p>
</div>
